options.go

```diff
@@ -162,6 +163,34 @@ func WithUniqueKeys(uniqueKeys ...bool) Option {
 	}
 }
 
+// WithStreamNodes enables returning stream boundary nodes when loading YAML.
+//
+// When enabled, Loader.Load returns an interleaved sequence of StreamNode and
+// DocumentNode values:
+//
+//	[StreamNode, DocNode, StreamNode, DocNode, ..., StreamNode]
+//
+// StreamNodes contain metadata about the stream including:
+//   - Encoding (UTF-8, UTF-16LE, UTF-16BE)
+//   - YAML version directive (%YAML)
+//   - Tag directives (%TAG)
+//   - Position information (Line, Column)
+//
+// An empty YAML stream returns a single StreamNode.
+// When called without arguments, defaults to true.
+//
+// The default is false.
+func WithStreamNodes(enable ...bool) Option {
+	if len(enable) > 1 {
+		panic("yaml: WithStreamNodes accepts at most one argument")
+	}
+	val := len(enable) == 0 || enable[0]
+	return func(o *options) error {
+		o.streamNodes = val
+		return nil
+	}
+}
```

Comment:
I would use this to avoid panic

```suggestion
func WithStreamNodes(enable ...bool) Option {
	var err error
	if len(enable) > 1 {
		err = errors.New("yaml: WithStreamNodes accepts at most one argument")
	}
	val := len(enable) == 0 || enable[0]
	return func(o *options) error {
		o.streamNodes = val
		return err
	}
}
```